cmd/cache_serve.go

```diff
@@ -146,43 +127,44 @@ func serveCacheInternal(ctx context.Context) error {
 	}
 
 	// Set up necessary APIs to support Web UI, including auth and metrics
-	if err := web_ui.ConfigureServerWebAPI(ctx, engine, egrp); err != nil {
+	if err := web_ui.ConfigureServerWebAPI(cmdCtx, engine, egrp); err != nil {
 		return err
 	}
 
-	go func() {
-		if err := web_ui.RunEngine(ctx, engine, egrp); err != nil {
-			log.Panicln("Failure when running the web engine:", err)
+	egrp.Go(func() error {
+		if err := web_ui.RunEngine(cmdCtx, engine, egrp); err != nil {
+			log.Errorln("Failure when running the web engine:", err)
+			return err
+		} else {
+			return err
 		}
-		shutdownCancel()
```

Comment:
This cancel (and the equivalent one in launchers) needs to be kept around.

The web engine may fail to startup and this `shutdownCancel()` call may be the thing that triggers the shutdown originally.

Reply:
@haoming29 Waiting on @bbockelm to answer this before I do my review, as this was my only concern as well.